### ISSUE TITLE
Cancellable speech: when going from one Explorer window to another, the title is discarded

### ISSUE THREAD

**Issue by leonardder** [nvaccess/nvda#11741]
### Steps to reproduce:
1. With cancellable speech enabled, open two Windows Explorer windows side by side, so you can alt+tab between them
2. Alt+tab between the windows
3. Close one window with alt+f4 in such a way that it brings you to the other window.

### Actual behavior:
For point 2: The title of the window is discared as soon as you release alt+tab. However, with rapid alt+tabbing, this means you'll miss the title
For point 3: THe title is also discared

### Expected behavior:
In both cases, the title of the Explorer window is discared

### Suggested fix
NVDA should never discard focus announcement messages for the foreground object.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-21145

#### Windows version:
10 (20h2) Build 19042.546

#### Name and version of other software in use when reproducing the issue:
N/a

#### Other information about your system:
N/a

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

#### If addons are disabled, is your problem still occuring?
Yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
N/a

**Comment by feerrenrut** [comment]
Interesting, I can't reproduce either problem you mention with `alpha-21145`.

Note that as of the #11632 change (delivered into build alpha-21139 and later) this does consider objects that are the foreground object. The full logic is:
``` python
	def _checkIfValid(self):
		stillValid = (
			self.isLastFocusObj()
			or not self.previouslyHadFocus()
			or self.isAncestorOfCurrentFocus()
			# Ensure titles for dialogs gaining focus are reported, EG NVDA Find dialog
			or self.isForegroundObject()
		)
		return stillValid
```

**Comment by leonardder** [comment]
Well, it gets even more interesting, as currently, I can't reproduce 
this either.

I will see how this behaves in the next week and will report back.

**Comment by lukaszgo1** [comment]
@leonardder How is this with the last Alpha?

**Comment by feerrenrut** [comment]
I think we should consider this fixed. @leonardder I'm happy for this to be re-opened if you can reproduce it.

**Comment by leonardder** [comment]
I can still reproduce this from time to time, however it is pretty hard to trigger this reliably. It disappears when I restart NVDA and than it takes a while to return.

**Comment by feerrenrut** [comment]
Are you sure it is related to cancellable speech?

**Comment by leonardder** [comment]
Yes. As soon as I turn it off, the issue disappears.

**Comment by leonardder** [comment]
I think I found the cause. It is related to cancellable speech in that it is only triggered when cancellable speech is enabled, but the source seems to be a running tree interceptor for UIA browse mode in Word that isn't killed correctly and generates a silenced error when alt+tabbing.

I'll investigate further.

**Comment by lukaszgo1** [comment]
@leonardder Can you still reproduce this?

**Comment by leonardder** [comment]
This was fixed by #11885, though I"m not sure about the underlying cause why this was related to cancellable speech. I don't think it's worth investigating further.
